src/ypd/model/project.py

```diff
@@ -115,7 +117,7 @@ def toggle_archived(self, user, session=None):
             self.archived = True
 
         else:
-            if user.is_admin:
+            if user.is_admin or self.can_be_modified_by(user):
```

Comment:
This is redundant.
If we only want admins to edit, we should use `if user.is_admin:` otherwise, it should be `if self.can_be_modified_by(user):`